### PR TITLE
feat: BL-17031 Update PHP to 8.2

### PR DESCRIPTION
**Pull request by UndermountainK**
## Description

**REVERTING CHANGES FOR VERSION 3.0.0** as its not working with MTS

Based on [PR 2 - php 8.1 update](https://github.com/dvsa/mot-cpms-client/pull/2)

Static analysis will be re-added in a later PR based from merged branch feat-bl-15039-add-static-analysis-php-8.2 (do not delete this branch) 

Related issue: [BL-17031](https://dvsa.atlassian.net/browse/BL-17031)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code?
- [ ] Have you have added tests that prove the fix or feature is effective and working?
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,20 +4,20 @@
     "type": "project",
     "license": "proprietary",
     "require": {
-        "php": "^8.0",
-        "dvsa/mot-cpms-notifications": "^1.0.0",
+        "php": "^8.2",
+        "dvsa/mot-cpms-notifications": "dev-feature-bl-17032-php-82-update",
```
Comment (mateuszc-kainos) [inline composer.json]: Create a tag in dependency repos and use it here. You can create a tag before merging to master, and after merging you can correct it if needed.

**Review by mateuszc-kainos — CHANGES_REQUESTED** [review]

**Review by mateuszc-kainos — DISMISSED** [review]

File: composer.json (hunk 1 above)
Comment (alexisc-kainos) [reply composer.json]: As discussed with @aaronlong , we will create tags once the swimlane is ready

**Review by alexisc-kainos — DISMISSED** [review]

**Review by alexisc-kainos — APPROVED** [review]

File: captainhook.json [hunk 2]
```diff
@@ -21,15 +21,6 @@
             {
                 "action": "composer test",
                 "options": []
-            },
```
Comment (waringr) [inline captainhook.json:24]: These need putting back in, along with the scripts and dependencies noted below. They run linting and static analysis on commit, to reduce the feedback loop.

File: composer.json [hunk 3]
```diff
@@ -5,27 +5,23 @@
...
-        "laminas/laminas-component-installer": "^3.4",
+        "laminas/laminas-component-installer": "^3.4.0",
         "laminas/laminas-test": "^4.0.0",
         "captainhook/captainhook": "^5.16",
         "captainhook/plugin-composer": "^5.3",
-        "friendsofphp/php-cs-fixer": "^3.17",
-        "vimeo/psalm": "^5.24",
-        "phpstan/phpstan": "^1.11",
-        "squizlabs/php_codesniffer": "^3.7",
-        "dvsa/coding-standards": "^2.0"
```
Comment (waringr) [inline composer.json:28]: As per above comment, these need putting back in

File: composer.json [hunk 4]
```diff
@@ -40,11 +36,7 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "scripts" : {
-        "test": "vendor/bin/phpunit -c phpunit.xml",
-        "phpcs": "./vendor/bin/phpcs .",
-        "phpcs-fix": "./vendor/bin/phpcbf .",
-        "psalm": "./vendor/bin/psalm" ,
-        "phpstan": "vendor/bin/phpstan analyse"
```
Comment (waringr) [inline composer.json:47]: And these are the scripts referred to in my top comment.

File: composer.json [hunk 5]
```diff
@@ -5,27 +5,23 @@
...
-        "dvsa/mot-cpms-notifications": "dev-feature-bl-17032-php-82-update",
+        "dvsa/mot-cpms-notifications": "^3.0.0",
         "laminas/laminas-mvc": "^3.3.0",
-        "laminas/laminas-cache": "^3.10.1",
+        "laminas/laminas-cache": "^3.12.0",
         "laminas/laminas-log": "^2.13",
         "laminas/laminas-filter": "^2.13",
         "laminas/laminas-dependency-plugin": "^2.6.0",
-        "laminas/laminas-cache-storage-adapter-filesystem": "^2.3.0",
-        "laminas/laminas-cache-storage-adapter-memory": "^2.2.0",
```
Comment (waringr) [inline composer.json:15]: Are these versions going backwards?

File: captainhook.json (hunk 2 above)
Comment (alexisc-kainos) [reply captainhook.json:24]: We've reverted these because we aren't running linting for this merge

File: phpcs.xml [hunk 6]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0"?>
-<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="mot-cpms-client" xsi:noNamespaceSchemaLocation="./vendor/squizlabs/php_codesniffer/phpcs.xsd">
```
Comment (waringr) [inline phpcs.xml:2]: I think this will need flipping back once the scripts etc. are re-added

File: phpcs.xml [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="mot-cpms-client" xsi:noNamespaceSchemaLocation="./vendor-bin/php-codesniffer/vendor/squizlabs/php_codesniffer/phpcs.xsd">
 
     <file>./src</file>
     <file>./test</file>
 
     <exclude-pattern>*/vendor/*</exclude-pattern>
 
-    <rule ref="./vendor/dvsa/coding-standards/src/Profiles/DVSA/CS/ruleset.xml" />
```
Comment (waringr) [inline phpcs.xml:9]: As the comment above

File: phpstan.neon [hunk 8]
```diff
@@ -1,6 +1,6 @@
 parameters:
-    ignoreErrors:
-        - identifier: missingType.iterableValue
+    checkMissingIterableValueType: false
```
Comment (waringr) [inline phpstan.neon:2]: This needs reverting to the original to fix a phpstan warning

File: psalm.xml
Comment (waringr) [inline psalm.xml:1]: This file will need reverting too.

File: captainhook.json (hunk 2 above)
Comment (waringr) [reply captainhook.json:24]: Ok, is there a plan to put them back in straight afterwards?

**Review by mateuszc-kainos — APPROVED** [review]

File: captainhook.json (hunk 2 above)
Comment (alexisc-kainos) [reply captainhook.json:24]: yes, we will probably raise tickets to fix linting and add strict types and add to the backlog (Aaron depending for priority)

File: composer.json (hunk 4 above)
Comment (alexisc-kainos) [reply composer.json:47]: Yes, these will work once we add them back in but we will leave these for now

File: composer.json (hunk 5 above)
Comment (alexisc-kainos) [reply composer.json:15]: We are just rebasing back to master version

File: phpcs.xml (hunk 6 above)
Comment (alexisc-kainos) [reply phpcs.xml:2]: Yeah we will add scripts in a new PR

File: phpcs.xml (hunk 7 above)
Comment (alexisc-kainos) [reply phpcs.xml:9]: yep

File: phpstan.neon (hunk 8 above)
Comment (alexisc-kainos) [reply phpstan.neon:2]: yes, we're not running the phpstan checks but this will be re-added in the next PR

**Pull request merged**